lib/main.dart

```diff
@@ -1,68 +1,21 @@
+import 'package:async_redux/async_redux.dart';
 import 'package:flutter/material.dart';
+import 'package:pokedex/pokedex_app.dart';
+import 'package:pokedex/state/app_state.dart';
+import 'package:pokedex/state/persistor.dart';
 
-void main() {
-  runApp(const MyApp());
-}
-
-class MyApp extends StatelessWidget {
-  const MyApp({Key? key}) : super(key: key);
-
-  @override
-  Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        primarySwatch: Colors.blue,
-      ),
-      home: const MyHomePage(title: 'Flutter Demo Home Page'),
-    );
-  }
-}
-
-class MyHomePage extends StatefulWidget {
-  const MyHomePage({Key? key, required this.title}) : super(key: key);
-
-
-  final String title;
-
-  @override
-  State<MyHomePage> createState() => _MyHomePageState();
-}
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+  final persistor = AppPersistor();
+  var initialState = await persistor.readState();
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+  final store = Store<AppState>(
+    initialState: initialState,
+    persistor: persistor,
+    actionObservers: [ConsoleActionObserver<AppState>()],
 
-  void _incrementCounter() {
-    setState(() {
-      _counter++;
-    });
-  }
+    /// TODO: this should be not present on the release mode
+  );
 
-  @override
-  Widget build(BuildContext context) {
-    return Scaffold(
-      appBar: AppBar(
-        title: Text(widget.title),
-      ),
-      body: Center(
-        child: Column(
-          mainAxisAlignment: MainAxisAlignment.center,
-          children: <Widget>[
-            const Text(
-              'You have pushed the button this many times:',
-            ),
-            Text(
-              '$_counter',
-              style: Theme.of(context).textTheme.headline4,
-            ),
-          ],
-        ),
-      ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: _incrementCounter,
-        tooltip: 'Increment',
-        child: const Icon(Icons.add),
-      ),
-    );
-  }
+  runApp(PokedexApp(store: store));
```

Comment:
do you really need to inject immediately the store here? 

Instead of passing it through, perhaps you can already provide the store by calling `StoreProvider` here and inject your PokedexApp, which will serve as your returned widget for running the app (i.e. `runApp(StoreProvider<AppState>(store: store, child: const PokedexApp()));`) 🙂 

Besides, you'll be setting-up later your go_router config in your PokedexApp, in there you will provide independent routers for you tab pages which will then callout the StoreConnector for further utilization 🙂